vendor/build-logger/src/ldlogger-tool-gcc.c

```diff
@@ -278,20 +341,46 @@ int loggerGccParserCollectActions(
   else  /* Compiler was not found in path, log the binary name only. */
   	  loggerVectorAdd(&action->arguments, loggerStrDup(toolName_));
 
+  /* Determine programming language based on compiler name. */
```

Comment:
In CodeChecker def get_language(extension).
in option_parser.py we determine the programming language based on the file extentsion.
It well may be that someone compiles a file with a g++ compiler in c mode.

get_language(extension):
    mapping = {'.c': 'c',
               '.cp': 'c++',
               '.cpp': 'c++',
               '.cxx': 'c++',
               '.txx': 'c++',
               '.cc': 'c++',
               '.C': 'c++',
               '.ii': 'c++',
               '.m': 'objective-c',
               '.mm': 'objective-c++'}
    lang = mapping.get(extension) if extension in mapping else None
    return lang

Of course the -x switch should override this choice, but wouldn't it be better to detect the language in the same way?